lib/object.c

```diff
@@ -42,14 +203,89 @@ static tool_rc tpm2_util_object_load2(ESYS_CONTEXT *ctx, const char *objectstr,
     }
 
     // 1. Always attempt file
+    tool_rc rc = tool_rc_general_error;
     FILE *f = fopen(objectstr, "rb");
     if (f) {
-        tool_rc rc = do_ctx_file(ctx, objectstr, f, outobject);
+        rc = do_ctx_file(ctx, objectstr, f, outobject);
         fclose(f);
+        if (rc != tool_rc_success) {
+            /*
+             * 2. if not parent context, then try for PEM
+             * Start by converting file to a openssl BIO object
+             */
+            BIO *input_bio = BIO_new_file(objectstr, "rb");
+            if (!input_bio) {
+                LOG_ERR("Unable to read as BIO file");
+                return tool_rc_general_error;
+            }
+
+            // fetch out the various parts of the PEM file using openssl API
```

Comment:
This style is for inline comment.
Here and elsewhere Use this instead:
/* 
 *
 */